starter-code/database.js

```diff
@@ -24,105 +23,105 @@ class Database {
     this.connect((error, database) => {
-      if (error){
+      if (error) {
         callback(error);
       } else {
         //  LAB 2
         // Implement the query to insert a user
         // remeber once it's finish to comment callback('Error listing users');
-        
-        callback('Error listing users');
+        database.collection('users').find({}, callback);
+        // callback('Error listing users');
       }
     });
   }
 
-  deleteUser( firstName, callback = (error, result) => {}) {
+  deleteUser (firstName, callback = (error, result) => {}) {
     this.connect((error, database) => {
-      if (error){
+      if (error) {
         callback(error);
       } else {
         //  LAB 3
         // Implement the query to delete a user
         // firstName is the name of user that we want to delete
         // remeber once it's finish to comment callback('Error deleting user');
-        
-        callback('Error deleting user');
+        database.collection('users').remove({firstName}, callback);
+        // callback('Error deleting user');
       }
     });
   }
 
-  insertProduct(product, callback = (error, result) => {}){
+  insertProduct (product, callback = (error, result) => {}) {
     this.connect((error, database) => {
-      if (error){
+      if (error) {
         callback(error);
       } else {
         // LAB 4
         // Implement the query to insert a product
         // product is the document to insert
         // remeber once it's finish to comment callback('Error inserting product');
-        
-        callback('Error inserting product');
+        database.collection('products').insertOne(product, callback);
+        // callback('Error inserting product');
       }
     });
   }
 
-  listProducts(callback = (error, products) => {}) {
+  listProducts (callback = (error, products) => {}) {
     this.connect((error, database) => {
-      if (error){
+      if (error) {
         callback(error);
       } else {
         // LAB 5
         // Implement the query to list all products
         // remeber once it's finish to comment callback('Error listing products');
-        
-        callback('Error listing products');
+        database.collection('products').find({}, callback);
+        // callback('Error listing products');
       }
     });
   }
 
-  deleteProduct( productName, callback = (error, result) => {}) {
+  deleteProduct (name, callback = (error, result) => {}) {
     this.connect((error, database) => {
-      if (error){
+      if (error) {
         callback(error);
       } else {
         // LAB 6
         // Implement the query to delete a product
-        // productName is the name of the producto to delete 
+        // productName is the name of the producto to delete
         // remeber once it's finish to comment callback('Error deleting product');
-        
-        callback('Error deleting product');
+        database.collection('products').remove({name}, callback);
```

Comment:
And here we have a problem, because the object key and the variable have different names, then you need to do:
```
database.collection('products').deleteOne('name' : productName, callback);
```
Your code will work because you have changed the parameter name (line 109). Obviously, you need to know this alternative. Ask me if this is not clear.